File: Py/problem_1.py
import requests
import time
from bs4 import BeautifulSoup
from pymongo import MongoClient

# Connect to MongoDB
mongo_client = MongoClient('mongodb://localhost:27017/')
db = mongo_client['stock_data']
stock_collection = db['nyse_stocks']

def fetch_stock_data():
    url = 'https://finance.yahoo.com/most-active'
    headers = {
        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36'
    }

    try:
        response = requests.get(url, headers=headers)
        if response.status_code == 200:
            soup = BeautifulSoup(response.content, 'html.parser')
            stocks_table = soup.find('table', {'class': 'W(100%)'})
            
            for row in stocks_table.find_all('tr')[1:]: 
                columns = row.find_all('td')
                stock_symbol = columns[0].text.strip()
                stock_name = columns[1].text.strip()
                stock_price = columns[2].text.strip()
                stock_change = columns[3].text.strip()
                stock_volume = columns[6].text.strip()

                
                stock_data = {
                    'Symbol': stock_symbol,
                    'Name': stock_name,
                    'Price': stock_price,
                    'Change': stock_change,
                    'Volume': stock_volume
                }
                stock_collection.insert_one(stock_data)
        else:
            print("Failed to fetch data. Status code:", response.status_code)
    except requests.RequestException as e:
        print("Request error:", e)

while True:
    fetch_stock_data()
    time.sleep(180)  

File: PHP/problem_1.php
<?php
require 'vendor/autoload.php'; 
?>

<!DOCTYPE html>
<html>
<head>
    <title>NYSE Stocks</title>
</head>
<body>
    <h1>Most Active Stocks on NYSE</h1>

    <table id="stockTable">
        <thead>
            <tr>
                <th>Symbol</th>
                <th>Name</th>
                <th>Price (Intraday)</th>
                <th>Change</th>
                <th>Volume</th>
            </tr>
        </thead>
        <tbody>
            <?php
            
            $mongoClient = new MongoDB\Client("mongodb://localhost:27017");
            $collection = $mongoClient->stock_data->nyse_stocks;

            
            $cursor = $collection->find();

            
            foreach ($cursor as $document) {
                echo '<tr>';
                echo '<td>' . $document['Symbol'] . '</td>';
                echo '<td>' . $document['Name'] . '</td>';
                echo '<td>' . $document['Price'] . '</td>';
                echo '<td>' . $document['Change'] . '</td>';
                echo '<td>' . $document['Volume'] . '</td>';
                echo '</tr>';
            }
            ?>
        </tbody>
    </table>

    
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
    
    <script src="https://cdn.datatables.net/1.11.5/js/jquery.dataTables.min.js"></script>
    
    <script>
        $(document).ready(function() {
            
            $('#stockTable').DataTable();
        });
    </script>
</body>
</html>
